ocaml/xapi/xmlm_tree.ml

```diff
@@ -0,0 +1,27 @@
+(** Conveniently output an XML tree using Xmlm *)
+
+(** Type used for constructing XML documents *)
+type tree =
+  | El of Xmlm.tag * tree list
+  | Data of string
+
+
+(** {2 Constructor functions} *)
+
+let data s = Data s
+
+(** Create an element without a namespace or attributes *)
+let el name children = El ((("", name), []), children)
+
+
+(** {2 Outputting a {!tree} using [Xmlm]} *)
+
+(** Creates a function deconstructing the above tree structure that can be
+    passed to Xmlm's tree output functions. *)
+let frag = function
+  | El (tag, children) -> `El (tag, children)
+  | Data s -> `Data s
+
+(** Outputs the given tree using [Xmlm.output_doc_tree] *)
+let output_doc ~tree ~output ~dtd =
+  Xmlm.output_doc_tree frag output (dtd, tree)
```

Comment:
I've inlined this in the SR.probe 1 PR in response to a review comment, so this file should be removed from the PR

Reply:
Thanks, dropped the file now